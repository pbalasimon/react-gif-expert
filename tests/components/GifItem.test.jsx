import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("<GifItem />", () => {
  const title = "This is the title";
  const url = "http://...";
  test("should match the snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should render the img with the url and alt", () => {
    render(<GifItem title={title} url={url} />);

    const { src, alt } = screen.getByRole("img");
    expect(src).toContain(url);
    expect(alt).toBe(title);
  });

  test("should show the title", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
