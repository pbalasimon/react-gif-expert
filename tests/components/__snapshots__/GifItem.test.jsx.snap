// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem /> should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="This is the title"
      src="http://..."
    />
    <p>
      This is the title
    </p>
  </div>
</div>
`;
